### DISCUSSION TITLE
How to JUCE + Elementary + React.js?

### DISCUSSION THREAD

**Discussion by satelllte** [elemaudio/elementary#18] [Q&A]
Hi @nick-thompson 👋🏻 

There's a video called "Building A Plugin With Elementary Audio" (https://youtu.be/vgq5fyxjSZE?t=511) where you're demonstrating the workflow with JUCE (as the base framework for VST / AU), Elementary Audio (for DSP) and React.js (for UI).

I'm wondered if there's a repo with this example available somewhere? I couldn't find it unfortunately.

More specifically I'm interested on how to start using React.js for UI in JUCE-based plugin? This would be so helpful to have an ability to have this 🙏🏻

**Comment by nick-thompson** [comment]
> There's a video called "Building A Plugin With Elementary Audio" (https://youtu.be/vgq5fyxjSZE?t=511) where you're demonstrating the workflow with JUCE (as the base framework for VST / AU), Elementary Audio (for DSP) and React.js (for UI).

Yes! In that talk I demonstrated the `plugin-renderer` of Elementary v1. That renderer is the one deprecated as of v2.0 and in it's place I have the [effects-plugin](https://github.com/elemaudio/effects-plugin) example repo. Currently the example repo does not include the web UI bindings, I'm still working on adding those along with a blog post explaining some of the more nuanced details, but the interaction between Elementary<>JUCE is already demonstrated there.

**Reply by satelllte** [reply]
Thank you! I saw that repo and I'm looking into it right now on how's the things work there. 

But for React.js, are you going to add some example for it anytime soon?

**Reply by nick-thompson** [reply]
> But for React.js, are you going to add some example for it anytime soon?

That's the goal, yep!
